### PR TITLE
Send all errors to Sentry

### PR DESCRIPTION
**Pull request by greenbigfrog**

### CODE REVIEW

File: src/discordtipbot/middleware/errorcatcher.cr [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class ErrorCatcher
+  def call(payload, context)
+    yield
+  rescue ex
+    Raven.capture(Exception.new("Exception while handling message #{payload}", ex))
```
Comment (z64) [inline src/discordtipbot/middleware/errorcatcher.cr]: Don't forget, you can be pretty creative with these, like letting the user know why their command was ignored, or didn't complete:

```cr
rescue ex
  ctx[Discord::Client].create_message(payload.channel_id, "Sorry, something unexpected went wrong!")
  Raven.capture(Exception.new("Exception while handling message #{payload}", ex))
```

**Comment by z64** [comment]
🏓 when rebased when #104 is merged. Should be g2g as is.

File: src/discordtipbot/middleware/errorcatcher.cr [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class ErrorCatcher
+  def call(payload, context)
+    yield
+  rescue ex
+    context[Discord::Client].create_message(payload.channel_id, "There was an unexpected error. This has been reported and should be resolved soon") if payload.is_a?(Discord::Gateway::MessageUpdatePayload)
```
Comment (z64) [inline src/discordtipbot/middleware/errorcatcher.cr]: .. `if payload.is_a?(Discord::Gateway::MessageUpdatePayload)` ? :confused:

Comment (greenbigfrog) [reply src/discordtipbot/middleware/errorcatcher.cr]: I want to call ErrorCatcher on all event handlers. Not only on_message, which is the only event it makes sense to reply to.

**Pull request merged**